### PR TITLE
chore(add-test): Added excoveralls, travis, credo and dialyxir

### PR DESCRIPTION
**Pull request by shraddha1704**
### What
* Added excoveralls, travis, credo and dialyxir

### Why
* Excoveralls for test coverage, Travis for open source testing, Credo and Dialyxir for code consistency and finding software discrepancies.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+language: elixir
+elixir: '1.6.5'
```
Comment (mpiercy827) [inline .travis.yml]: Do we want to support different versions of elixir as well? Or are we just starting at `1.6.5`?

File: mix.exs [hunk 2]
```diff
@@ -10,6 +10,7 @@ defmodule Litmus.MixProject do
       elixir: "~> 1.6",
       start_permanent: Mix.env() == :prod,
       deps: deps(),
+      dialyzer: [plt_add_deps: :apps_direct, plt_add_apps: [:wx]],
```
Comment (mpiercy827) [inline mix.exs]: what is the `:wx` for?

File: mix.exs [hunk 3]
```diff
@@ -41,7 +51,10 @@ defmodule Litmus.MixProject do
   # Run "mix help deps" to learn about dependencies.
   defp deps do
     [
-      {:ex_doc, "~> 0.18.0", only: :dev, runtime: false}
+      {:ex_doc, "~> 0.18.0", only: :dev, runtime: false},
+      {:credo, "~> 0.9.1", only: [:dev, :test], runtime: false},
```
Comment (mpiercy827) [inline mix.exs:54]: Are we missing a `credo.exs`?

File: mix.exs [hunk 4]
```diff
@@ -27,6 +28,15 @@ defmodule Litmus.MixProject do
...
           "GitHub" => @github_url,
         }
+      ],
+
+      #ExCoveralls
+      test_coverage: [tool: ExCoveralls],
+      preferred_cli_env: [
+        "coveralls": :test,
+        "coveralls.detail": :test,
+        "coveralls.post": :test,
```
Comment (mpiercy827) [inline mix.exs]: We can probably leave out the `.detail` and `.post` lines? Unless there is a good reason to keep them in.

File: .travis.yml (hunk 1 above)
Comment (mpiercy827) [reply .travis.yml]: Let's also add [spaces between the lines](https://github.com/lob/lob-elixir/blob/master/.travis.yml).

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+otp_release:
+  - '20.0'
+  - '20.1'
+  - '20.2'
+  - '20.3'
```
Comment (mpiercy827) [inline .travis.yml]: I think we'll also need to specify some build steps as well. I think `mix compile`, `mix credo` and `mix dialyzer` could be good for now.

File: mix.exs (hunk 4 above)
Comment (mpiercy827) [reply mix.exs]: But we should add `coveralls.travis`, since we are using Travis CI.

**Review by mpiercy827 — CHANGES_REQUESTED** [review]
Couple of changes

File: mix.exs (hunk 3 above)
Comment (shraddha1704) [reply mix.exs:54]: added it

File: .travis.yml (hunk 5 above)
Comment (shraddha1704) [reply .travis.yml]: added them in the yml file

File: mix.exs [hunk 6]
```diff
@@ -41,7 +50,10 @@ defmodule Litmus.MixProject do
   # Run "mix help deps" to learn about dependencies.
   defp deps do
     [
-      {:ex_doc, "~> 0.18.0", only: :dev, runtime: false}
+      {:ex_doc, "~> 0.18.0", only: :dev, runtime: false},
+      {:credo, "~> 0.9.1", only: [:dev, :test], runtime: false},
+      {:excoveralls, "~> 0.8", only: :test},
+      {:dialyxir, "~> 0.5", only: [:dev], runtime: false}
```
Comment (mpiercy827) [inline mix.exs]: we can remove the `[]` here, to be consistent with the other dependencies.

Comment (shraddha1704) [reply mix.exs]: added test mode to dialyxir

**Review by mpiercy827 — DISMISSED** [review]
LGTM, pending the Travis Build.

File: .travis.yml [hunk 7]
```diff
@@ -0,0 +1,23 @@
+language: elixir
+
+sudo: false
+
+elixir:
```
Comment (shraddha1704) [inline .travis.yml:5]: Will add other versions if needed in future

**Review by mgartner — DISMISSED** [review]
LGTM

**Review by mgartner — DISMISSED** [review]

File: .credo.exs [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+        {Credo.Check.Design.TagTODO, exit_status: 2},
+        {Credo.Check.Design.TagFIXME},
+
+        #
+        ## Readability Checks
+        #
+        {Credo.Check.Readability.AliasOrder},
+        {Credo.Check.Readability.FunctionNames},
+        {Credo.Check.Readability.LargeNumbers},
+        {Credo.Check.Readability.MaxLineLength, priority: :low, max_length: 80},
```
Comment (kylekwong) [inline .credo.exs]: Let's make this 98 to match the [formatter's default max length](https://github.com/christopheradams/elixir_style_guide/issues/56#issuecomment-385379132)

**Review by kylekwong — DISMISSED** [review]

File: .credo.exs (hunk 8 above)
Comment (shraddha1704) [reply .credo.exs]: sure

**Review by kylekwong — APPROVED** [review]

**Review by mpiercy827 — APPROVED** [review]

**Pull request merged**